File: task2.js
function task2(numbers) {
    if (!numbers.length) {
      return [];
    }
  
    function gcd(a, b) {
      while (b) {
        [a, b] = [b, a % b];
      }
      return a;
    }
  
    let scm = numbers[0];
    for (let i = 1; i < numbers.length; i++) {
      scm = gcd(scm, numbers[i]);
    }
  
    return Array.from({ length: scm - 1 }, (_, i) => i + 2).filter(num => scm % num === 0);
  }

File: task2.py
def task2(numbers:list)->list:
    if not numbers:
      return []

    def gcd(a, b):
        while b:
            a, b = b, a % b
        return a
    
    scm = numbers[0]
    for number in numbers[1:]:
        scm = gcd(scm, number)

    return [ num for num in range(2, scm + 1) if scm % num == 0]